### PR TITLE
Graphite

### PR DESCRIPTION
**Pull request by apeschar**
###### Motivation for this change

* Graphite 1.0.2 is newer.
* This also fixes carbonRelay and carbonAggregate services, that specified a PIDFile in a non-existent RuntimeDirectory.
* `django_tagging_0_3` was only used by the older Graphite, so can be removed. (I suppose.)

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
    - Made sure all the services work
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -25665,21 +25657,18 @@ EOF
 
     src = pkgs.fetchurl rec {
       url = "mirror://pypi/g/graphite-web/${name}.tar.gz";
-      sha256 = "1c0kclbv8shv9nvjx19wqm4asia58s3qmd9fapchc6y9fjpjax6q";
+      sha256 = "0q8bwlj75jqyzmazfsi5sa26xl58ssa8wdxm2l4j0jqyn8xpfnmc";
     };
 
-    propagatedBuildInputs = with self; [ django django_tagging_0_3 whisper pycairo ldap memcached pytz ];
+    propagatedBuildInputs = with self; [ django_1_8 (django_tagging.override { django = django_1_8; }) whisper pycairo ldap memcached pytz urllib3 cairocffi scandir pyparsing ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: No overriding within the package set. All Django packages will need to depend just on `django`, and not on a specific version. If you application needs a specific version, then you need to override the whole Python package set when deploying it.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (apeschar) [inline pkgs/top-level/python-packages.nix]: When deploying it -> you mean in the service?

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: yes

**Comment by apeschar** [comment]
@FRidh:

* graphite_web now uses just django and django_tagging
* django is overridden in the pythonPackages that is used in the service

This means the 'stock' pythonPackages.graphite_web doesn't build, since it relies on Django 1.8. Is that expected?

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Comment by FRidh** [comment]
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

**Comment by FRidh** [comment]
@apeschar yes, that can be fine.

**Comment by apeschar** [comment]
This was already done in another pull request, so no longer relevant.

**Pull request closed without merging**